### ISSUE TITLE
Incorrect Color Behavior

### ISSUE THREAD

**Issue by Radnen** [spheredev/neosphere#12]
So, I don't know what Sphere is doing, but this should make the Rectangle rotate between black and white as if the colors were modulus 255.

``` javascript
var d = 0;
var c = CreateColor(d, d, d);
while (true) {
    c.red = d;
    c.green = d;
    c.blue = d;

    Rectangle(0, 0, GetScreenWidth(), GetScreenHeight(), c);
    FlipScreen();
    d++;
}
```

It does this correctly in SSFML because I used modulus on the color, but here's where it gets weird.

``` javascript
var d = 255 + 127;
var c = CreateColor(d, d, d);
Rectangle(0, 0, GetScreenWidth(), GetScreenHeight(), c);
FlipScreen();
GetKey();
```

This produces White in Sphere 1.5 and minisphere, but grey in SSFML. That means it's not as simple as color modulus. But still, the colors were rotating in Sphere 1.5 with the first code example. I don't know if it is basic byte overflow... Because if it were, then you'd get weird results with example #2... so I'm not sure here.

**Comment by fatcerberus** [comment]
minisphere explicitly and deliberately clamps colors to [0-255].  I did this because the modulo-like behavior messes with tweening.

Here's where I think the discrepancy is: CreateColor in Sphere clamps to [0-255], the same as minisphere.  However, it doesn't clamp direct assignments to the components, hence the rotation behavior in the first snippet, which can be directly attributed to overflow.

**Comment by Radnen** [comment]
Hmm, don't know then if that breaks anything? I like your implementation, it seems like an honest bug fix over Sphere 1.5.

Plus, why would you have colors that take 300 for a channel when 255 is the max? No other graphics library treats colors like that, this is only an issue when you go between statically typed and dynamically typed languages like this. I'd say this is alright.

**Comment by fatcerberus** [comment]
It's a fine line to walk with things like this.  When it's an obvious bug I'll fix it, even if that means breaking things that rely on the bugged behavior. 

Sometimes though, like here, it gets ambiguous, and then you have to use your best judgment.  In this case the output of CreateColor can reasonably be considered canonical, and then the overflow on increment can be treated as a bug.
